File: ggbackend/src/main/java/net/goodgreek/ggbackend/model/Event.java
package net.goodgreek.ggbackend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "event")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "organization")
    private String organization;

    @Column(name = "eventName") 
    private String eventName;

    @Column(name = "eventDescription")
    private String eventDescription;

    @Column(name = "location")
    private String location;

    @Column(name = "date")
    private String date;

    @Column(name = "price") 
    private String price;

    @Column(name = "venmo") 
    private String venmo;

    @Column(name = "benefits")
    private String benefits;

    @Column(name = "website")
    private String website;

    @Column(name = "image")
    private int image;
}

File: Front-end/goodgreek/src/pages/Event.js
import React, { useEffect, useState } from "react";
import "../css/App.css";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import '../css/event.css';
import sample from '../assets/sampleposter.jpg';
import axios from 'axios';
import EventsService from "../services/EventsService";

import donation from '../assets/donation.png';
import unity from '../assets/unity.jpg';
import volunteer from '../assets/volunteer.jpeg';
import hearthands from '../assets/hearthands.jpg';

import "bootstrap/dist/css/bootstrap.min.css";

const Event = () => {
  const [eventData, setEventData] = useState(null);
  const [loading, setLoading] = useState(true);

  const images = {donation, unity, volunteer, hearthands};

  const id = new URLSearchParams(window.location.search).get("id");

  useEffect(() => {
    try {
      EventsService.getEventById(id).then((response) => {
        setEventData(response.data);
        setLoading(false);
      });
    }
    catch(err) {
      setLoading(false);
    }
    
  }, [id]);

  if (loading) {
    return <div className="whitetext">Loading...</div>;
  }

  if (!eventData) {
    return <div className="whitetext">No event found</div>;
  }

  const { eventName, organization, date, location, eventDescription, price, venmo, benefits, website, image } = eventData;

  return (
    <div className="event">
      <image src={images[image]}></image>
      <div className="event-info">
        <h1>Name: {eventName}</h1>
        <p>Organizer: {organization}</p>
        <p>Date: {date}</p>
        <p>Location: {location}</p>
        {price && <p>Price: {price}</p>}
        {venmo && <p>Venmo: {venmo}</p>}
        {benefits && <p>Benefits: {benefits}</p>}
        {website && <p>Charity Website: {website}</p>}
        <p>Description: {eventDescription}</p>
      </div>
      
    </div>
  );
};

export default Event;